src/libraries/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -86,6 +86,24 @@ public static bool TryGet(ReadOnlySpan<byte> headerName, out HeaderDescriptor de
             return true;
         }
 
+        internal static bool TryGetStaticQPackHeader(int index, out HeaderDescriptor descriptor, out string knownValue)
+        {
+            Debug.Assert(index >= 0);
+            Debug.Assert(s_qpackHeaderLookup.Length == 99);
+
+            if (index < s_qpackHeaderLookup.Length)
+            {
+                (descriptor, knownValue) = s_qpackHeaderLookup[index];
```

Comment:
I don't know whether the JIT is going to special-case the static readonlyness here, but you might want to store this into a local.  If you do that and cast to uint in the condition, I believe it'll remove the bounds check, e.g.
```C#
(HeaderDescriptor descriptor, string value)[] lookup = s_qpackHeaderLookup;
if ((uint)index < lookup)
{
    (descriptor, knownValue) = lookup[index];
    return true;
}
...
```
Of course it might not matter at all.

Reply:
I'll check codegen, good callout.

Reply:
Confirmed, using a local prevents re-reaching Length from memory. Applying both of these.